File: app-miyou/src/main/java/com/miyou/app/application/credit/usecase/CreditDeductUseCase.java
package com.miyou.app.application.credit.usecase;

import com.miyou.app.domain.credit.model.CreditTransaction;
import com.miyou.app.domain.dialogue.model.ConversationSessionId;
import com.miyou.app.domain.dialogue.model.UserId;
import reactor.core.publisher.Mono;

public interface CreditDeductUseCase {

	Mono<CreditTransaction> deductForConversation(UserId userId, ConversationSessionId sessionId);
}

File: app-miyou/src/main/kotlin/com/miyou/app/application/credit/usecase/CreditDeductUseCase.kt
package com.miyou.app.application.credit.usecase

import com.miyou.app.domain.credit.model.CreditTransaction
import reactor.core.publisher.Mono

interface CreditDeductUseCase {
    fun deductForConversation(
        userId: String,
        sessionId: String,
    ): Mono<CreditTransaction>

    fun refundForConversation(
        userId: String,
        sessionId: String,
    ): Mono<CreditTransaction>
}
